sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,196 @@
+
+        files = self.aws_manager.get_bucket_structure_with_prefix("shadowrobot.benchmarks", self._hand_serial)
+        if files is None:
+            return ""
+
+        files.sort(key=lambda x: x['LastModified'])
+        last_file_path = files[-1]['Key']
+        latest_file_name_from_AWS = last_file_path[last_file_path.find("/")+1:]
+
+        return latest_file_name_from_AWS
+
+    def _upload_to_AWS(self, event):
+        success = False
+
+        orignal_file = self._log_file_path + self._log_file_name
+        now = datetime.datetime.now()
+        dated_file = now.strftime("%Y-%m-%d-%H-%M-%S") + ".yaml"
+        shutil.copy(orignal_file, self._log_file_path + dated_file)
+        rospy.sleep(1)
+
+        while os.stat(self._log_file_path + dated_file).st_size == 0:
+            shutil.copy(orignal_file, self._log_file_path + dated_file)
+            rospy.sleep(0.5)
+
+        success = self.aws_manager.upload("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                          self._hand_serial, [dated_file])
+        rospy.sleep(1)
+
+        if os.path.exists(self._log_file_path + dated_file):
+            os.remove(self._log_file_path + dated_file)
+
+        return success
+
+    def _download_from_AWS(self):
+        success = False
+        success = self.aws_manager.download("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                            self._hand_serial, [self._get_latest_file_name_from_AWS()])
+        if success:
+            rospy.sleep(1)
+            latest_file_name_from_AWS = self._get_latest_file_name_from_AWS()
+            shutil.copy(self._log_file_path + latest_file_name_from_AWS, self._log_file_path + self._log_file_name)
+            rospy.sleep(1)
+            if os.path.exists(self._log_file_path + latest_file_name_from_AWS):
+                os.remove(self._log_file_path + latest_file_name_from_AWS)
+
+        return success
+
+    def _save_data_localy(self, event):
+        if self._verify_data_empty():
+            print("saving data locally")
+            self._complete_data['total_angles_[rad]'] = self._current_values
+            self._complete_data['total_time_[s]'] = rospy.get_rostime().secs
+            f = open(self._log_file_path+self._log_file_name, 'w')
+            yaml.safe_dump(self._complete_data, f)
+            f.close()
+
+    def _verify_data_empty(self):
+        if len(self._current_values.keys()) == 0:
+            return False
+        return True
+
+    def _load_data_from_yaml(self):
+        with open(self._log_file_path+self._log_file_name, 'r') as f:
+            self._complete_data = yaml.load(f, Loader=yaml.SafeLoader)
+        self._current_values = self._complete_data['total_angles_[rad]']
+        self._current_time = self._complete_data['total_time_[s]']
+
+    def _update_with_higher_values(self, local_data, aws_data):
+        for k, v in local_data.items():
```

Comment:
we should probably protect against empy local data?

Reply:
Corrected